.circleci/config.templ.yml

```diff
@@ -0,0 +1,1153 @@
+          pattern: 'yaaredis$'
+          snapshot: true
+
+  redis:
+    <<: *machine_executor
+    steps:
+      - run_test:
+          docker_services: 'redis rediscluster'
+          pattern: 'redis$'
+          snapshot: true
+
+  rediscluster:
+    <<: *machine_executor
+    steps:
+      - run_test:
+          pattern: 'rediscluster'
+          docker_services: 'rediscluster'
+          snapshot: true
+
+  rq:
+    <<: *machine_executor
+    parallelism: 2
+    steps:
+      - run_test:
+          pattern: "rq"
+          snapshot: true
+          docker_services: "redis"
+
+  urllib3:
+    <<: *machine_executor
+    steps:
+      - run_test:
+          pattern: 'urllib3'
+          snapshot: true
+          docker_services: "httpbin_local"
+
+  vertica:
+    <<: *contrib_job
+    docker:
+      - image: *ddtrace_dev_image
+      - *testagent
+      - image: *vertica_image
+        environment:
+          - VP_TEST_USER=dbadmin
+          - VP_TEST_PASSWORD=abc123
+          - VP_TEST_DATABASE=docker
+    steps:
+      - run_test:
+          wait: vertica
+          pattern: 'vertica'
+
+  wsgi:
+    <<: *machine_executor
+    steps:
+      - run_test:
+          pattern: "wsgi"
+          snapshot: true
+
+  kafka:
+    <<: *machine_executor
+    parallelism: 2
+    steps:
+      - run_test:
+          pattern: 'kafka'
+          snapshot: true
+          docker_services: 'kafka'
+
+  kombu:
+    <<: *contrib_job
+    docker:
+      - image: *ddtrace_dev_image
+      - image: *rabbitmq_image
+      - *testagent
+    steps:
+      - run_test:
+          wait: rabbitmq
+          pattern: 'kombu'
+
+  jinja2:
+    <<: *contrib_job_small
+    docker:
+      - image: *ddtrace_dev_image
+      - *testagent
+    steps:
+      - run_test:
+          pattern: 'jinja2'
+
+  mako:
+    <<: *contrib_job
+    docker:
+      - image: *ddtrace_dev_image
+      - *testagent
+    steps:
+      - run_test:
+          pattern: 'mako'
+
+  algoliasearch:
+    <<: *contrib_job
+    docker:
+      - image: *ddtrace_dev_image
+      - *testagent
+    steps:
+      - run_test:
+          pattern: 'algoliasearch'
+
+  langchain:
+    <<: *machine_executor
+    steps:
+      - run_test:
+          pattern: "langchain"
+          snapshot: true
+
+  build_docs:
+    # build documentation and store as an artifact
+    executor: ddtrace_dev
+    steps:
+      - setup_riot
+      - checkout
+      - run:
+          command: |
+             riot -v run docs
+             mkdir -p /tmp/docs
+             cp -r docs/_build/html/* /tmp/docs
+      - store_artifacts:
+          path: /tmp/docs
+
+requires_pre_check: &requires_pre_check
+  requires:
+    - pre_check
+    - ccheck
+
+requires_base_venvs: &requires_base_venvs
+  requires:
+    - pre_check
+    - ccheck
+    - build_base_venvs
+
+requires_tests: &requires_tests
+  requires:
+#[[[cog from needs_testrun import for_each_testrun_needed as fetn;fetn(lambda job: print(f"    - {job}")) ]]]
```

Comment:
This generates the list of required jobs for `requires_test`. All the jobs are listed in the `needs_testrun.py` script now. This helps DRY this config source a bit since we can list them once, instead of repeating them in the workflow jobs below.